bigtable-dataflow-import/src/main/java/com/google/cloud/bigtable/dataflowimport/HBaseResultToMutationFn.java

```diff
@@ -48,25 +81,78 @@ static void setLogger(Logger log) {
   @Override
   public void processElement(ProcessContext context) throws Exception {
     KV<ImmutableBytesWritable, Result> kv = context.element();
-    Cell[] cells = checkEmptyRow(kv);
-    if (cells.length == 0) {
+    List<Cell> cells = checkEmptyRow(kv);
+    if (cells.isEmpty()) {
       return;
     }
-    // TODO: Handle DeleteMarkers properly.
-    Put put = new Put(kv.getKey().get());
+    Put put = tryProcessRowWithNoDeleteMarkers(kv.getKey().get(), cells);
```

Comment:
I don't know if this is worth the effort to do two tries :(
can we do this?
Put put = new Put(rowKey);
Map<String, Collection<Cell>> deleteMarkersByFamilyMap = ...
for (...)  {
     if (IS_DELETE_MARKER_FILTER.apply(cell) {
       deleteMarkersByFamilyMap.put(...);
     } else {
       put.add(cell);
     }
}

if (deleteMarkersByFamilyMap.isEmpty())  {
  return put;
}

// remove deleted cells.